Modules/Loadable/Segmentations/Testing/Python/SegmentationWidgetsTest1.py

```diff
@@ -77,3 +76,216 @@ def TestSection_1_qMRMLSegmentsTableView(self):
     self.assertEqual(len(segmentsTableView.displayedSegmentIDs()), 2)
     slicer.app.processEvents()
     slicer.util.delayDisplay("Hidden 'second'")
+
+  #------------------------------------------------------------------------------
+  def compareOutputGeometry(self, orientedImageData, spacing, origin, directions):
+    if orientedImageData is None:
+      logging.error('Invalid input oriented image data')
+      return False
+    if (not isinstance(spacing, list) and not isinstance(spacing, tuple)) \
+        or (not isinstance(origin, list) and not isinstance(origin, tuple)) \
+        or not isinstance(directions, list):
+      logging.error('Invalid baseline object types - need lists')
+      return False
+    if len(spacing) != 3 or len(origin) != 3 or len(directions) != 3 \
+        or len(directions[0]) != 3 or len(directions[1]) != 3 or len(directions[2]) != 3:
+      logging.error('Baseline lists need to contain 3 elements each, the directions 3 lists of 3')
+      return False
+    import numpy
+    tolerance = 0.0001
+    actualSpacing = orientedImageData.GetSpacing()
```

Comment:
spacing, orientation, and directions comparison could be simplified by using 

    static bool vtkAddonMathUtilities::MatrixAreEqual(const vtkMatrix4x4* m1, const vtkMatrix4x4* m2, double tolerance = 1e-3);

Reply:
I know. However I wanted to be able to log error messages that are more punctual and are easier to interpret.

Reply:
OK. If this comes up again then we may create a shared helper function in vtkAddon or slicer.util that can provide these more descriptive messages.